### PR TITLE
Refined security procedure names and namespace

### PR DESCRIPTION
**Pull request by craigtaverner**
- Change addUserToRole -> assignRoleToUser
- Change removeUserFromRole -> removeRoleFromUser
- Separate namespace for new security procedures (ie. `dbms.security.*`)

### CODE REVIEW

File: community/security/src/main/java/org/neo4j/server/security/auth/AuthProcedures.java [hunk 1]
```diff
@@ -34,7 +34,14 @@
     @Context
     public AuthSubject authSubject;
 
+    @Deprecated
     @Procedure( name = "dbms.changePassword", mode = DBMS )
```
Comment (fickludd) [inline community/security/src/main/java/org/neo4j/server/security/auth/AuthProcedures.java]: `deprecatedBy = dbms.security.changePassword`?

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTestLogic.java [hunk 2]
```diff
@@ -533,137 +537,138 @@ public void shouldTerminateConnectionsOnUserSuspension() throws Exception
...
 
     @Test
     public void shouldFailToAddNonExistentUserToRole() throws Exception
     {
-        testFailAddUserToRole( adminSubject, "Olivia", PUBLISHER, "User 'Olivia' does not exist." );
-        testFailAddUserToRole( adminSubject, "Olivia", "thisRoleDoesNotExist", "User 'Olivia' does not exist." );
-        testFailAddUserToRole( adminSubject, "Olivia", "", "The provided role name is empty." );
+        testFailAddRoleToUser( adminSubject, PUBLISHER, "Olivia", "User 'Olivia' does not exist." );
+        testFailAddRoleToUser( adminSubject, "thisRoleDoesNotExist", "Olivia", "User 'Olivia' does not exist." );
+        testFailAddRoleToUser( adminSubject, "", "Olivia", "The provided role name is empty." );
```
Comment (fickludd) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresTestLogic.java:598]: Awesome, you change the test methods too. Stringent!

**Comment by fickludd** [comment]
Looks good to me.

File: community/security/src/main/java/org/neo4j/server/security/auth/AuthProcedures.java (hunk 1 above)
Comment (craigtaverner) [reply community/security/src/main/java/org/neo4j/server/security/auth/AuthProcedures.java]: Support for deprecatedBy is in another PR, so we need to add that once the other PR is in (probably a small followup PR with a one-line addition)

**Comment by craigtaverner** [comment]
Rebased on top of work to move user management to community

**Pull request merged**